pkg/launcher/launcher.go

```diff
@@ -61,6 +61,7 @@ func (launcher *Launcher) Run() error {
 			"exitCode": err,
 			"cmd":      cmdStr,
 		}).Debug("Process exited")
+		exitCallback()
```

Comment:
You probably want to wrap the call with `if existCallback != nil { ... }`, as in some places the method is called with `exitCallback=nil`.